src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/zk/ZooKeeperBolt.java

```diff
@@ -81,6 +83,7 @@ protected void handleInput(Tuple input) throws Exception {
     @Override
     protected void init() {
         initZk();
+        zkWriter.setExpectedState(expectedState);
```

Comment:
I'd prefer to have this as a constructor arg, and just override initAndWaitConnection in writer with this value recording in zk